File: language/src/main/java/brj/nodes/builtins/NowNode.java
package brj.nodes.builtins;

import brj.BridjeLanguage;
import brj.nodes.ExprNode;
import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
import com.oracle.truffle.api.frame.VirtualFrame;

public class NowNode extends ExprNode {

    public NowNode(BridjeLanguage lang) {
        super(lang);
    }

    @TruffleBoundary
    private long currentTimeMillis() {
        return System.currentTimeMillis();
    }

    @Override
    public Object execute(VirtualFrame frame) {
        return currentTimeMillis();
    }
}

File: language/src/main/kotlin/brj/builtins/NowNode.kt
package brj.builtins

import brj.BridjeLanguage
import brj.nodes.ExprNode
import brj.runtime.BridjeInstant
import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary
import com.oracle.truffle.api.frame.VirtualFrame
import java.time.Instant

class NowNode(lang: BridjeLanguage) : ExprNode(lang) {

    @TruffleBoundary
    private fun now() = BridjeInstant(Instant.now())

    override fun execute(frame: VirtualFrame?): Any = now()
}